drivers/gpio/gpio_fxl6408.c

```diff
@@ -0,0 +1,438 @@
+				   value);
+}
+
+static inline int update_pul_en_regs(const struct device *dev, uint8_t value)
+{
+	struct gpio_fxl6408_drv_data *const drv_data =
+		(struct gpio_fxl6408_drv_data *const)dev->data;
+
+	return write_port_regs(dev, REG_PUD_EN, &drv_data->reg_cache.pud_en,
+				   value);
+}
+
+static int setup_pin_dir(const struct device *dev, uint32_t pin, int flags)
+{
+	struct gpio_fxl6408_drv_data *const drv_data =
+		(struct gpio_fxl6408_drv_data *const)dev->data;
+	uint8_t reg_dir = drv_data->reg_cache.dir;
+	uint8_t reg_out = drv_data->reg_cache.output;
+	uint8_t reg_high_z = drv_data->reg_cache.high_z;
+	int ret;
+
+	/* Update the driver data to the actual situation of the FXL6408 */
+	if (flags & GPIO_OUTPUT) {
+		if ((flags & GPIO_OUTPUT_INIT_HIGH)) {
+			reg_out |= BIT(pin);
+		} else if ((flags & GPIO_OUTPUT_INIT_LOW)) {
+			reg_out &= ~BIT(pin);
+		}
+		reg_dir |= BIT(pin);
+		reg_high_z &= ~BIT(pin); 
+	} else if (flags & GPIO_INPUT) {
+		reg_dir &= ~BIT(pin);
+		reg_high_z &= ~BIT(pin); 
+	} else {
+		reg_high_z |= BIT(pin); 
+	}
+
+	ret = update_output_regs(dev, reg_out);
+	if (ret != 0) {
+		return ret;
+	}
+	
+	ret = update_high_z_regs(dev, reg_high_z);
+	if (ret != 0) {
+		return ret;
+	}
+
+	ret = update_direction_regs(dev, reg_dir);	
+	return ret;
+}
+
+/**
+ * @brief Setup the pin pull up/pull down status
+ *
+ * @param dev Device struct of the FXL6408
+ * @param pin The pin number
+ * @param flags Flags of pin or port
+ *
+ * @return 0 if successful, failed otherwise
+ */
+static int setup_pin_pullupdown(const struct device *dev, uint32_t pin,
+				int flags)
+{
+	struct gpio_fxl6408_drv_data *const drv_data =
+		(struct gpio_fxl6408_drv_data *const)dev->data;
+	uint8_t reg_pud;
+	int ret;
+
+	/* If disabling pull up/down, there is no need to set the selection
+	 * register. Just go straight to disabling.
+	 */
+	if ((flags & (GPIO_PULL_UP | GPIO_PULL_DOWN)) != 0U) {
+		/* Setup pin pull up or pull down */
+		reg_pud = drv_data->reg_cache.pud_sel;
+
+		/* Pull down == 0, pull up == 1 */
+		WRITE_BIT(reg_pud, pin, (flags & GPIO_PULL_UP) != 0U);
+
+		ret = update_pul_sel_regs(dev, reg_pud);
+		if (ret) {
+			return ret;
+		}
+	}
+
+	/* Enable/disable pull up/down */
+	reg_pud = drv_data->reg_cache.pud_en;
+
+	WRITE_BIT(reg_pud, pin,
+		  (flags & (GPIO_PULL_UP | GPIO_PULL_DOWN)) != 0U);
+
+	ret = update_pul_en_regs(dev, reg_pud);
+	return ret;
+}
+
+static int gpio_fxl6408_config(const struct device *dev, gpio_pin_t pin,
+				   gpio_flags_t flags)
+{
+	struct gpio_fxl6408_drv_data *const drv_data =
+		(struct gpio_fxl6408_drv_data *const)dev->data;
+	int ret;
+
+	/* Check if supported flag is set */
+	if (!(flags & SUPPORTED_FLAGS)) {
+		return -ENOTSUP;
+	}
+
+	/* Can't do I2C bus operations from an ISR */
+	if (k_is_in_isr()) {
+		return -EWOULDBLOCK;
+	}
+	k_sem_take(&drv_data->lock, K_FOREVER);
+
+	ret = setup_pin_dir(dev, pin, flags);
+	if (ret) {
```

Comment:
Let's write it as `(ret != 0)`, here and in a few more places, to use the same style. This is recommended (but not yet enforced) but the coding guidelines.